types/lodash/index.d.ts

```diff
@@ -6366,98 +6366,34 @@ declare namespace _ {
         /**
          * @see _.forEach
          */
-        each<T>(
-            collection: T[],
-            iteratee?: ListIterator<T, any>
-        ): T[];
-
-        /**
-         * @see _.forEach
-         */
-        each(
-            collection: string,
+        each<TString extends string | null | undefined>(
+            collection: TString,
             iteratee?: ListIterator<string, any>
-        ): string;
+        ): TString;
 
         /**
          * @see _.forEach
          */
-        each<T>(
-            collection: List<T>,
+        each<T, TList extends List<T> | null | undefined>(
```

Comment:
@aj-r 
Hm... It's because `Array<number>` subtype of `Array<{}>`. TypeScript can't narrow its type further...
Using `TList extends List<T>` itself make the problem, so you're right, this change should be reverted until type reference is more stronger.